### PR TITLE
fix: Make examples sequencer compatible with modern TBB

### LINKED ISSUE #823: Make examples Sequencer work with newer versions of TBB (2021.x)
The examples' Sequencer uses the `tbb::task_scheduler_init` API, which has been deprecated for a while and has been fully removed in newer TBB 2021.x versions. We should investigate what the replacement is and change the Sequencer's code so that it works with newer TBB. Let's cross fingers that we won't need an ifdef mess to stay compatible with older TBB...

### PR DESCRIPTION
**Pull request by HadrienG2**
Fixes #823 .

### CODE REVIEW

File: Examples/Framework/CMakeLists.txt [hunk 1]
```diff
@@ -19,8 +19,8 @@ target_include_directories(
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
 target_link_libraries(
   ActsExamplesFramework
-  PUBLIC ActsCore ActsFatras Boost::boost ROOT::Core ROOT::Hist
-  PRIVATE Boost::filesystem dfelibs TBB::tbb)
+  PUBLIC ActsCore ActsFatras Boost::boost ROOT::Core ROOT::Hist TBB::tbb
```
Comment (HadrienG2) [inline Examples/Framework/CMakeLists.txt:22]: Needed to call a class member's destructor.

Comment (paulgessinger) [reply Examples/Framework/CMakeLists.txt:22]: Ok

File: Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp [hunk 2]
```diff
@@ -22,6 +22,8 @@
 #include <utility>
 #include <vector>
 
+#include "tbb/task_arena.h"
```
Comment (paulgessinger) [inline Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp:25]:
```suggestion
#include <tbb/task_arena.h>
```

**Review by paulgessinger — APPROVED** [review]
Cool, so this should work with 2020 and 2021 versions. Neat.

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**